src/gtc/dace/expansion.py

```diff
@@ -580,62 +683,202 @@ def get_innermost_memlets(self):
         access_collection: AccessCollector.CartesianAccessCollection = get_access_collection(
             self.node
         )
-
+        variable_k_accesses = {
+            acc.name
+            for acc in self.node.oir_node.iter_tree().if_isinstance(oir.FieldAccess)
+            if isinstance(acc.offset, common.VariableKOffset)
+        }
+        dynamic_accesses = {
```

Comment:
There seem to be a number of cases covered here in the unions. Could you add a comment to describe the cases covered? There could also be some work done to remove code duplication.

Reply:
We discussed this and will resolve in the next iteration of the expansion.